web-common/src/features/chat/core/messages/thinking/ThinkingBlock.svelte

```diff
@@ -0,0 +1,133 @@
+  import type { V1Message, V1Tool } from "../../../../../runtime-client";
+  import { MessageType } from "../../types";
+  import AnimatedDots from "../AnimatedDots.svelte";
+  import ToolCall from "../tools/ToolCall.svelte";
+
+  export let messages: V1Message[];
+  export let resultMessagesByParentId: Map<string | undefined, V1Message>;
+  export let isComplete: boolean;
+  export let duration: number;
+  export let tools: V1Tool[] | undefined = undefined;
+
+  let isExpanded = true;
+  let hasUserInteracted = false;
+
+  // Auto-collapse when thinking completes, unless user has interacted
+  $: if (isComplete && !hasUserInteracted) {
+    isExpanded = false;
+  }
+
+  function formatDuration(seconds: number): string {
+    if (seconds < 1) return "less than a second";
+    if (seconds === 1) return "1 second";
+    if (seconds < 60) return `${seconds} seconds`;
+
+    const minutes = Math.floor(seconds / 60);
+    const remainingSeconds = seconds % 60;
+
+    if (remainingSeconds === 0) {
+      return minutes === 1 ? "1 minute" : `${minutes} minutes`;
+    }
+
+    return `${minutes} ${minutes === 1 ? "minute" : "minutes"} ${remainingSeconds} ${remainingSeconds === 1 ? "second" : "seconds"}`;
+  }
+
+  function toggleExpanded() {
+    hasUserInteracted = true;
+    isExpanded = !isExpanded;
+  }
+</script>
+
+<div class="thinking-block">
+  <button class="thinking-header" on:click={toggleExpanded}>
+    <div class="thinking-icon">
+      {#if isExpanded}
+        <CaretDownIcon size="14" color="currentColor" />
+      {:else}
+        <Brain />
+      {/if}
+    </div>
+    <div class="thinking-title">
+      {#if isComplete}
+        Thought for {formatDuration(duration)}
+      {:else}
+        <AnimatedDots>Thinking</AnimatedDots>
+      {/if}
+    </div>
+  </button>
+
+  {#if isExpanded}
+    <div class="thinking-messages">
+      {#each messages as msg (msg.id)}
+        {#if msg.type === MessageType.PROGRESS}
+          {#if msg.contentData}
+            <div class="thinking-content">
+              <Markdown content={msg.contentData} />
+            </div>
+          {/if}
+        {:else if msg.type === MessageType.CALL}
+          <ToolCall
+            message={msg}
+            resultMessage={resultMessagesByParentId.get(msg.id)}
+            {tools}
+            variant="inline"
+          />
+        {/if}
+      {/each}
+    </div>
+  {/if}
+</div>
+
+<style lang="postcss">
+  .thinking-block {
+    @apply w-full max-w-full self-start;
+  }
+
+  .thinking-header {
+    @apply w-full flex items-center gap-1.5 py-1;
+    @apply bg-transparent border-none cursor-pointer;
+    @apply text-xs text-gray-500 transition-colors;
+  }
+
+  .thinking-header:hover {
+    @apply text-gray-600;
+  }
+
+  .thinking-icon {
+    @apply flex items-center;
+  }
+
+  .thinking-title {
+    @apply flex-1 text-left font-normal;
+  }
+
+  .thinking-messages {
+    @apply pl-5 flex flex-col gap-0;
+    @apply max-h-96 overflow-y-auto;
```

Comment:
Is it by design to have this? On mac since scroll bar doesnt appear without a scroll it was not apparent for me that there are more items. Since there is scroll outside for the whole history section I thought this would just take as much space as available. Not sure if there is a good way to make this clear.